pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCache.java

```diff
@@ -144,6 +172,17 @@ public String getActualTableName(String tableName) {
     }
   }
 
+  /**
+   * Returns the actual logical table name for the given table name, or {@code null} if table does not exist.
+   * @param logicalTableName Logical table name
+   * @return Actual logical table name
+   */
+  public String getActualLogicalTableName(String logicalTableName) {
```

Comment:
Is this function only used in tests ? No other use right now ? I checked the usages of `getActualTableName` and it is used in a bunch of places. Some of them are out of scope for this PR. Can you re-verify though ?

Reply:
Modified the `getActualTableName` in the `BaseSingleStageBrokerRequestHandler` to handle both logical and regular tables.